third_party/accessibility/ax/platform/ax_platform_node_textrangeprovider_win.cc

```diff
@@ -0,0 +1,1701 @@
+      // matches the attribute and its value encountered.
+      matched_range_start = current_start->Clone();
+      matched_range_end = current_end->Clone();
+    }
+  };
+
+  HRESULT hr_result =
+      is_backward
+          ? FindAttributeRange(text_attribute_id, attribute_val,
+                               anchors.crbegin(), anchors.crend(), expand_match)
+          : FindAttributeRange(text_attribute_id, attribute_val,
+                               anchors.cbegin(), anchors.cend(), expand_match);
+  if (FAILED(hr_result))
+    return E_FAIL;
+
+  if (matched_range_start != nullptr && matched_range_end != nullptr)
+    *result = CreateTextRangeProvider(std::move(matched_range_start),
+                                      std::move(matched_range_end));
+  return S_OK;
+}
+
+template <typename AnchorIterator, typename ExpandMatchLambda>
+HRESULT AXPlatformNodeTextRangeProviderWin::FindAttributeRange(
+    const TEXTATTRIBUTEID text_attribute_id,
+    VARIANT attribute_val,
+    const AnchorIterator first,
+    const AnchorIterator last,
+    ExpandMatchLambda expand_match) {
+  AXPlatformNodeWin* current_platform_node;
+  bool is_match_found = false;
+
+  for (auto it = first; it != last; ++it) {
+    const auto& current_start = it->anchor();
+    const auto& current_end = it->focus();
+
+    BASE_DCHECK(current_start->GetAnchor() == current_end->GetAnchor());
+
+    AXPlatformNodeDelegate* delegate = GetDelegate(current_start);
+    BASE_DCHECK(delegate);
+
+    current_platform_node = static_cast<AXPlatformNodeWin*>(
+        delegate->GetFromNodeID(current_start->GetAnchor()->id()));
+
+    base::win::VariantVector current_attribute_value;
+    if (FAILED(current_platform_node->GetTextAttributeValue(
+            text_attribute_id, current_start->text_offset(),
+            current_end->text_offset(), &current_attribute_value))) {
+      return E_FAIL;
+    }
+
+    if (!current_attribute_value.Compare(attribute_val)) {
+      // When we encounter an AXRange instance that matches the attribute
+      // and its value which we are looking for and no previously matched text
+      // range exists, we expand or initialize the matched range.
+      is_match_found = true;
+      expand_match(current_start, current_end);
+    } else if (is_match_found) {
+      // When we encounter an AXRange instance that does not match the attribute
+      // and its value which we are looking for and a previously matched text
+      // range exists, the previously matched text range is the result we found.
+      break;
+    }
+  }
+  return S_OK;
+}
+
+static bool StringSearch(const std::u16string& search_string,
+                         const std::u16string& find_in,
+                         size_t* find_start,
+                         size_t* find_length,
+                         bool ignore_case,
+                         bool backwards) {
+  // TODO(schectman) Respect ignore_case
```

Comment:
Do we also need to add internationalization support?

Reply:
Maybe. I do not know what actual use case calls this method.

Reply:
What do you think of updating the issue/comment to mention both internalization and case insensitivity?